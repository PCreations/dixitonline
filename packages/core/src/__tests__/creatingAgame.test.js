const uuid = require('uuid/v1');
const { EventEmitter } = require('events');
const { DixitCore } = require('../index');
const { reduceToDixitState } = require('../domain/state');

describe('Feature: creating a game', () => {
  describe('given the authenticated user being user42', () => {
    describe('and the lobby being empty', () => {
      describe('when user42 creates a new game', () => {
        test("then the lobby should contain the created game with user42 in player's list", () => {
          const eventEmitter = new EventEmitter();
          const sendCommand = command => eventEmitter.emit('command', command);
          const dispatchEvents = events => events.map(event => eventEmitter.emit('event', event));
          const consumeCommands = onCommand => eventEmitter.addListener('command', onCommand);
          const consumeEvents = onEvent => eventEmitter.addListener('event', onEvent);
          const getNextGameId = uuid;
          const inMemoryEventStore = [];
          let inMemoryState;
          const saveEvents = events => inMemoryEventStore.concat([events]);
          const dixitCore = DixitCore({
            consumeCommands,
            consumeEvents,
            dispatchEvents,
            saveEvents,
            getNextGameId,
            getState: () => inMemoryState,
            updateState: state => {
              inMemoryState = state;
            },
          });
          sendCommand(gameCommands.createGameCommand());
        });
      });
    });
  });
});
