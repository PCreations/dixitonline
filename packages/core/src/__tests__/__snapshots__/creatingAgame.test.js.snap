// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scenario: creating a game when the lobby is empty given the authenticated user being user42 and the lobby being empty when user42 creates a new game then the lobby should contain the created game with user42 in player's list: lobby should be empty 1`] = `
Object {
  "games": Object {},
}
`;

exports[`Scenario: creating a game when the lobby is empty given the authenticated user being user42 and the lobby being empty when user42 creates a new game then the lobby should contain the created game with user42 in player's list: lobby should contain the created game with the player "player42" as the only player 1`] = `
Object {
  "games": Object {
    "game42": Object {
      "id": "game42",
      "players": Array [
        Object {
          "id": "42",
          "name": "player42",
        },
      ],
    },
  },
}
`;

exports[`Scenario: creating a game when the lobby is empty given the authenticated user being user42 and the lobby being empty when user42 creates a new game then the lobby should contain the created game with user42 in player's list: lobby should only contains the new empty game 1`] = `
Object {
  "games": Object {
    "game42": Object {
      "id": "game42",
      "players": Array [],
    },
  },
}
`;
